rq/worker.py

```diff
@@ -387,6 +386,20 @@ def kill_horse(self, sig=SIGKILL):
             else:
                 raise
 
+    def wait_for_horse(self):
+        """
+        A waiting the end of the horse process and recycling resources.
+        """
+        try:
+            pid = None
+            stat = None
+            if self.horse_pid:
```

Comment:
No need to check for the existence of `horse_pid` since this method is only called by the main worker.

Reply:
These codes have been adjusted.